### PR TITLE
[Infra] Add support for handling unknown redirects outside of profiles

### PR DESCRIPTION
**Pull request by orta**
Was having a think about https://github.com/artsy/eigen/issues/1314 - this should fix it. Want to give it a proper think through, so not aiming to get it in this release.

/cc @alloy

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">No CHANGELOG changes made</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
  </tbody>
</table>

Time | Class | Function |
| --- | ----- | ----- |232.4ms | [AuctionViewController.swift](/artsy/eigen/tree/orta-handle_unknown_urls/Artsy/View_Controllers/Auction/AuctionViewController.swift#L183) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
173.9ms | [SwiftyJSON.swift](/artsy/eigen/tree/orta-handle_unknown_urls/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
158.0ms | [WebSocket.swift](/artsy/eigen/tree/orta-handle_unknown_urls/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
72.6ms | [Observable.swift](/artsy/eigen/tree/orta-handle_unknown_urls/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
66.7ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/orta-handle_unknown_urls/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, sideMargin: String)
65.5ms | [LiveAuctionsAdminViewController.swift](/artsy/eigen/tree/orta-handle_unknown_urls/Artsy/View_Controllers/Live_Auctions/LiveAuctionsAdminViewController.swift#L22) | @objc override func viewDidLoad()
59.2ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/orta-handle_unknown_urls/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L25) | func ar_presentBlurredOverlayWithTitle(title: String, subtitle: String, buttonState: BlurredStatusOverlayViewCloseButtonState = default)
57.7ms | [LiveAuctionLotViewController.swift](/artsy/eigen/tree/orta-handle_unknown_urls/Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift#L49) | @objc override func viewDidLoad()
57.4ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/orta-handle_unknown_urls/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L31) | (closure)
55.3ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/orta-handle_unknown_urls/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by alloy** [comment]
Looking forward to it 👍

File: Artsy/App/ARSwitchboard+Eigen.h [hunk 1]
```diff
@@ -69,4 +70,6 @@
 /// Buy artwork
 - (UIViewController *)loadOrderUIForID:(NSString *)orderID resumeToken:(NSString *)resumeToken;
 
+/// Crazy Artsy redirect things
```
Comment (alloy) [inline Artsy/App/ARSwitchboard+Eigen.h]: Would prefer a helpful comment :)

File: Artsy/View_Controllers/ARMutableLinkViewController.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+/// synchronously, so this VC wil do a head request to gravity
+/// to discover what the route after redirection is.
+///
+/// This new path will then be re-routed through ARRouter, resulting
+/// in either a ProfileVC or the real view controller for that path.
+///
+/// This is then added as a child view controller, or presented modally
+/// on the top menu view controller.
+
+
```
Comment (alloy) [inline Artsy/View_Controllers/ARMutableLinkViewController.h]: This is helpful 👍

So what happens when you navigate back from the real VC? Won't you get back to this MutableVC instance?

Comment (alloy) [inline Artsy/View_Controllers/ARMutableLinkViewController.h]: Ah no, I missed the part about adding as a child VC

File: Artsy/View_Controllers/ARMutableLinkViewController.m [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        if (responseCode == 302 && headers[@"Location"]) {
+            internalViewController = [ARSwitchBoard.sharedInstance loadPath:headers[@"Location"]];
+        } else {
+            internalViewController = [ARSwitchBoard.sharedInstance loadProfileWithID:self.originalPath];
+        }
+
+        if ([ARTopMenuViewController shouldPresentViewControllerAsModal:internalViewController]) {
+            [[ARTopMenuViewController sharedController] pushViewController:internalViewController animated:ARPerformWorkAsynchronously completion:^{
+                [self.navigationController popViewControllerAnimated:YES];
```
Comment (alloy) [inline Artsy/View_Controllers/ARMutableLinkViewController.m]: Why is this always animated? Is that correct?

**Comment by alloy** [comment]
Otherwise LGTM 👍

File: Artsy/View_Controllers/ARMutableLinkViewController.m (hunk 3 above)
Comment (orta) [inline Artsy/View_Controllers/ARMutableLinkViewController.m]: Shouldn't matter to my knowledge, considering it's behind the scenes, I bet that not animated is probably the better call

Comment (orta) [inline Artsy/View_Controllers/ARMutableLinkViewController.m]: To make sure the presentedVC doesn't accidentally end up in an animation context

**Pull request merged**